hub.go

```diff
@@ -67,60 +70,98 @@ func (hub *Hub) LastEventID() EventID {
 }
 
 func (hub *Hub) stackTop() *layer {
+	hub.RLock()
 	stack := hub.stack
 	if stack == nil || len(*stack) == 0 {
+		hub.RUnlock()
```

Comment:
I have no idea about GO, but can we maybe use `defer`?
So this is valid for every function, so we always do:
```
hub.RLock()
defer hub.RUnlock()
```